src/main/scala/com/lightbend/kafka/scala/streams/typesafe/TSTimeWindowedKStream.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018 OpenShine SL <https://www.openshine.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.lightbend.kafka.scala.streams.typesafe
+
+import com.lightbend.kafka.scala.streams.FunctionConversions._
+import com.lightbend.kafka.scala.streams.typesafe.ImplicitConverters._
+import org.apache.kafka.streams.kstream.{Materialized, TimeWindowedKStream,
+  Windowed}
+
+class TSTimeWindowedKStream[K, V]
+(protected[typesafe] override val unsafe: TimeWindowedKStream[K, V])
+  extends AnyVal with TSKType[TimeWindowedKStream, K, V] {
```

Comment:
Same as above:
I don't think this is a great to `extends AnyVal` here since this can lead to a lot of boxing and unboxing.
See: https://failex.blogspot.co.uk/2017/04/the-high-cost-of-anyval-subclasses.html